src/util/PriorityQueue.hxx

```diff
@@ -0,0 +1,62 @@
+#ifndef PRIORITY_QUEUE_HXX_
+#define PRIORITY_QUEUE_HXX_
+
+#include <vector>
+
+/**
+  * @brief Priority queue with removing functionality.
+  */
+template <typename T> class PriorityQueue
+{
+  std::vector<T> m_container;
+
+public:
+  using Iterator = typename std::vector<T>::iterator;
+
+  /**
+    * @return Iterator to beginning of queue.
+    */
+  Iterator begin() { return m_container.begin(); }
+
+  /**
+    * @return Iterator to end of queue.
+    */
+  Iterator end() { return m_container.end(); }
+
+  /**
+    * @brief Check whether queue is empty.
+    * @return true if queue is empty otherwise false.
+    */
+  bool isEmpty() { return m_container.empty(); }
+
+  /**
+    * @brief Get top element from the queue.
+    * @return Top element from queue.
+    */
+  T &top() { return m_container.front(); }
+
+  /**
+    * @brief Add new element to the queue.
+    */
+  void add(T &&element);
```

Comment:
I would rename this to `::push` to follow STL conventions, but it's ok. You should use proper typename & also support const reference
```suggestion
  void add(value_type &&element);
  void add(const value_type &element);
```

Reply:
Also you should use `std::move` for the rvalue overload

Reply:
Done